core/trino-main/src/main/java/io/trino/connector/ConnectorManager.java

```diff
@@ -165,8 +165,8 @@ public synchronized void stop()
 
         for (Map.Entry<CatalogName, MaterializedConnector> entry : connectors.entrySet()) {
             Connector connector = entry.getValue().getConnector();
-            try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(connector.getClass().getClassLoader())) {
-                connector.shutdown();
+            try {
+                withClassLoader(connector.getClass(), connector::shutdown);
```

Comment:
> `withClassLoader(connector.getClass(),`

this is wrong. a Class is not a ClassLoader.

Reply:
This override makes chained `getClassLoader()` calls unnecessary and just takes class-loader for a class:

```
public static void withClassLoader(Class<?> clazz, Runnable runnable)
{
    withClassLoader(clazz.getClassLoader(), runnable);
}
```

https://github.com/trinodb/trino/pull/8507/files#diff-d5f1fa216ac7b2b536a6fa79582157d1270bdf985b6901056e5074c355be2daeR20

Reply:
i know that it does that. But still, at the call site, it _reads wrong_.

Reply:
Naming things is hard. It does read wrong, I get your point. Maybe 'withClassLoader**Of**(Class<?>, ...)'?

Reply:
that would be better